### DISCUSSION TITLE
Onboarding PR not created

### DISCUSSION THREAD

**Discussion by kamilkowalski** [renovatebot/renovate#18969] [Request Help] [answered]
We're using the Renovate GitHub App on a private repository in the `surgeventures` organization.

Some time ago (not sure how long as we didn't check Renovate runs), Renovate stopped working on our `surgeventures/system` repository - all runs are marked as "Disabled".

I wanted to re-onboard this repository to the Renovate GitHub App so I:

1. Renamed the `renovate.json` file to `renovate-backup.json`
2. Since we didn't have an on-boarding PR, I renamed the "Dependency Dashboard" issue to "Dependency Dashboard - old"
3. Uninstalled and installed the app again

I expected the file rename to open an onboarding PR, but even the re-install didn't trigger it. Why is Renovate disabled on our repo and how do I re-enable it?

**Comment by rarkins** [comment, marked as answer]
Repos which permanently error out get disabled on our server side. I have removed this disablement now but you should try to ensure to address any problems such as caused by unreachable private dependencies, etc.

**Reply by kamilkowalski** [reply]
Thank you, I'll look into the issues. I do remember we had some problems with Hex private packages - I've implemented support for them in #11168 but while it worked on a test repo, it didn't work on our actual one. I'll reach out if that's still an issue, maybe other users have a similar one and we could fix it.

**Comment by kamilkowalski** [comment]
@rarkins so after enabling Renovate again on our repo and fixing some of the errors, I've arrived at two that aren't configuration issues (see run #898974183):

1. I'm getting some ECONNRESET errors when fetching both public Hex and npm packages - even after updating the timeout to 240000 as suggested in [this discussion](https://github.com/renovatebot/renovate/discussions/17646).
2. I see that our private package `ecto_refresh` is being fetched from the public repository, even though our `mix.exs` file specifies the `organization:` key - I believe [this regex](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/mix/extract.ts#L9) is the issue and assumes that `organization:` is the only configuration option.

I'll try to open a PR fixing 2, but do you have any clue about 1?

**Reply by rarkins** [reply]
Issue 1 has unfortunately been plaguing us for some time especially in very large repos. Something about the production environment leads to an increased number of ETIMEDOUT or ECONNRESET errors. We're not sure if it's better to increase the time out of instead decrease it while making sure we retry enough. Do the error messages you get indicate that any retries happened?

**Reply by kamilkowalski** [reply]
Yes, it seems so:

```
DEBUG: GET https://registry.npmjs.org/write-pkg = (code=ECONNRESET, statusCode=-1 retryCount=2, duration=32807)
```

All `ECONNRESET` errors have a `retryCount` of 2.

**Reply by kamilkowalski** [reply]
@rarkins I've [opened a PR for issue 2](https://github.com/renovatebot/renovate/pull/19128).

As to the connection issues - I don't see an option for retries, so do you suggest tweaking the timeout? Would running Renovate another way help? Is it an issue of too many requests from the same pool of IPs?

**Reply by rarkins** [reply]
I think if you switched to running Renovate self-hosted, e.g. using the GitHub Action, then the problem would likely disappear. For some reason it appears to be environmental, and we can't reproduce it locally. We don't know if it's our fault via either a Renovate code bug which only appears in production, which seems unlikely, or something to do with our production VMs. Alternatively maybe we are getting throttled somehow between us and GitHub/npm, because those are the only datasources which we have observed it happening to.
